### PR TITLE
fix: make extra call for getting ccc status

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=5W4B-Ij)
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Comment by suyashpatil78** [comment]
will fix tests later

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 1]
```diff
@@ -83,7 +86,53 @@ export class ExpensesService {
       },
     };
 
-    return this.spenderService.get<PlatformApiResponse<Expense[]>>('/expenses', data).pipe(map((res) => res.data[0]));
+    //return this.spenderService.get<PlatformApiResponse<Expense[]>>('/expenses', data).pipe(map((res) => res.data[0]));
```
Comment (Dimple16) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:89]: we can remove this and add a `TODO` to remove the condition once we get rid of slow sync in the backend

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 2]
```diff
@@ -93,9 +142,49 @@ export class ExpensesService {
   }
 
   getExpenses(params: ExpensesQueryParams): Observable<Expense[]> {
-    return this.spenderService
-      .get<PlatformApiResponse<Expense[]>>('/expenses', { params })
-      .pipe(map((expenses) => expenses.data));
+    return this.spenderService.get<PlatformApiResponse<Expense[]>>('/expenses', { params }).pipe(
+      map((expenses) => expenses.data),
+      switchMap((expenses) => {
+        const idsOfExpenseWithNoMatchedCCCTransactions = expenses
```
Comment (Dimple16) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:148]: we can move this block to a separate method - `getIdsOfExpenseWithNoMatchedCCCTxns()`

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 3]
```diff
@@ -93,9 +142,49 @@ export class ExpensesService {
...
+              expense.matched_corporate_card_transactions?.length === 0
+          )
+          .map((expense) => expense.matched_corporate_card_transaction_ids[0]);
+
+        if (idsOfExpenseWithNoMatchedCCCTransactions.length > 0) {
+          const params = {
+            id: `in.(${idsOfExpenseWithNoMatchedCCCTransactions.join(',')})`,
+          };
+
+          return this.spenderService
```
Comment (Dimple16) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:161]: same for the below block, we can keep fetching and mapping of ccc in a separate method and call it here
This way we can keep the CCC part logic outside the `getExpenses()` method

**Review by Dimple16 — APPROVED** [review]
added comments

**Review by OmkarJ13 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

**Pull request merged**